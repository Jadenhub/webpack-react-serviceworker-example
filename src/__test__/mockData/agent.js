export const MOBILE_AGENT = 'Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) OPiOS/10.2.0.93022 Mobile/11D257 Safari/9537.53'
export const DESKTOP_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246'