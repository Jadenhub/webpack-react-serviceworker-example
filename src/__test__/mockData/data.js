export const RATES = {
  "rates":{
    "btc": {
      "name":"Bitcoin",
      "unit":"BTC",
      "value":1.0,
      "type":"crypto"
    },
    "eth": {
      "name":"Ether",
      "unit":"ETH",
      "value":16.198,
      "type":"crypto"
    },
    "ltc": {
      "name":"Litecoin",
      "unit":"LTC",
      "value":230.349,
      "type":"crypto"
    },
    "bch": {
      "name":"Bitcoin Cash",
      "unit":"BCH",
      "value":64.588,
      "type":"crypto"
    },
    "bnb": {
      "name":"Binance Coin",
      "unit":"BNB",
      "value":105.539,
      "type":"crypto"
    },
    "eos": {
      "name":"EOS",
      "unit":"EOS",
      "value":8003.441,
      "type":"crypto"
    },
    "xrp": {
      "name":"XRP",
      "unit":"XRP",
      "value":47106.088,
      "type":"crypto"
    },
    "xlm": {
      "name":"Lumens",
      "unit":"XLM",
      "value":119836.321,
      "type":"crypto"
    },
    "link": {
      "name":"Chainlink",
      "unit":"LINK",
      "value":1693.022,
      "type":"crypto"
    },
    "dot":{
      "name":"Polkadot",
      "unit":"DOT",
      "value":1736.006,
      "type":"crypto"
    },
    "yfi":{
      "name":"Yearn.finance",
      "unit":"YFI",
      "value":0.985,
      "type":"crypto"
    }
  }
};
