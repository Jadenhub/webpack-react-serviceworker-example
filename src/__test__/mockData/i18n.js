export const I18N = {
  PROJECT_NAME: 'webapp',
  META_TITLE: 'Webapp',
  RATES: 'Rates',
  RATES_LIST: 'RATES LIST',
  NAME: 'Name',
  TYPE: 'Type',
  VALUE: 'Value',
  UNIT: 'Unit',
  SELECT_ALL: 'Select All',
  PAGE: 'Page',
  GO_TO_PAGE: 'Go to page',
  SHOW: 'Show',
  OF: 'of'
}