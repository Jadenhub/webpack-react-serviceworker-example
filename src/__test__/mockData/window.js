export function MockWindow(){
  global.window = {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn()
  }
}