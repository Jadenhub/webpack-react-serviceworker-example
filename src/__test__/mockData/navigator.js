import { DESKTOP_AGENT, MOBILE_AGENT } from "./agent";


export function MockDesktopNavigator(){
  global.navigator = {
    userAgent: DESKTOP_AGENT
  }
}

export function MockMobileNavigator(){
  global.navigator = {
    userAgent: MOBILE_AGENT
  }
}