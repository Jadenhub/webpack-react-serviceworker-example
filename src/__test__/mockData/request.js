import { DESKTOP_AGENT, MOBILE_AGENT } from "./agent"

export function getMockDesktopRequest(){
  return {
    path: '/',
    headers: {
      'user-agent': DESKTOP_AGENT
    }
  }
}

export function getMockMobileRequest(){
  return {
    path: '/',
    headers: {
      'user-agent': MOBILE_AGENT
    }
  }
}