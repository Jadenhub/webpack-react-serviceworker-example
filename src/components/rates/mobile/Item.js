import styled from 'styled-components';
import { withI18n } from "../../decorator/i18n";
import React from 'react';
import { HIGHLIGHT_COLOR, ITEM_DEFAULT_COLOR} from '../../../constants/style';

const RootWrapper = styled.div`
  border: 1px solid rgb(151, 151, 151);
  display: flex;
  flex-wrap: wrap;
  background-color: ${(props)=> props.clicked? HIGHLIGHT_COLOR: ITEM_DEFAULT_COLOR};
  box-shadow: 1px 1px 4px 1px rgba(114, 114, 114, 0.5);
  cursor:pointer;
`
const ItemWrapper = styled.div`
  width: 50%;
  padding: 10px;
  box-sizing: border-box;
`

function Item({
  rate, i18n, clicked, onClick
}){
  const {name, type, value, unit} = rate;
  return (
    <RootWrapper clicked={clicked} onClick={onClick}>
      <ItemWrapper>
        {i18n["NAME"]}:{name}
      </ItemWrapper>
      <ItemWrapper>
        {i18n["TYPE"]}:{type}
      </ItemWrapper>
      <ItemWrapper>
        {i18n["VALUE"]}:{value}
      </ItemWrapper>
      <ItemWrapper>
        {i18n["UNIT"]}:{unit}
      </ItemWrapper>
    </RootWrapper>
  )
}

export default withI18n(Item)