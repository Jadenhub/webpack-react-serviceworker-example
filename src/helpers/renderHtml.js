import React from 'react';
import { renderToString } from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { MOBILE } from '../constants/devices';
import Root from '../Root';
import { getTranslation } from './i18n';
import { getDeviceType } from './userAgent';

function getProps(props){
  return`
    <script>
      window.__PAGE_PROPS__ = ${JSON.stringify(props).replace(
        /</g,
        '\\u003c'
      )}
    </script>
  `
}

const render = async (req) => {
  const sheet = new ServerStyleSheet();
  try {
    const i18n = getTranslation();
    const isMobile = getDeviceType(req) === MOBILE;
    const content = renderToString(sheet.collectStyles(
      <Root
        isMobile={isMobile}
        req={req}
        i18n={i18n}
      />
    ));
    const styleTags = sheet.getStyleTags();
    const html = `
      <!DOCTYPE html>
      <html>
        <head>
          <link href="/favicon.ico" rel="icon">
          <title>${i18n['META_TITLE']}</title>
          <meta charSet='utf-8'>
          <meta content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no' name='viewport'>
          ${styleTags}
          <link rel="preload" href="/main.css" as="style"/>
          <link rel="stylesheet" href="/main.css">
        </head>
        <body>
          <div id='root'>${content}</div>
          ${getProps({i18n, isMobile})}
          <script src="/bundle.js"></script>
        </body>
      </html>
    `;
    return html;
  } catch (error) {
    console.error(error);
  } finally {
    sheet.seal();
  }
};

export default render;