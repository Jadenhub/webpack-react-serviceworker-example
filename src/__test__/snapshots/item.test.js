import renderer from 'react-test-renderer';
import React from 'react';
import { getTranslation } from '../../helpers/i18n';
import { RATES } from '../mockData/data';
import Item from '../../components/rates/mobile/Item';

test('renders correctly', () => {
  const tree = renderer
    .create(<Item 
      rate={RATES.rates.btc} 
      i18n={getTranslation()}
      clicked={true}
      onClick={()=>{}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});