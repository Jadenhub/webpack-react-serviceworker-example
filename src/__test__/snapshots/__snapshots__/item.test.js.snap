// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Item__RootWrapper-w49u1w-0 gPBKOw"
  onClick={[Function]}
>
  <div
    className="Item__ItemWrapper-w49u1w-1 joGTJV"
  >
    Name
    :
    Bitcoin
  </div>
  <div
    className="Item__ItemWrapper-w49u1w-1 joGTJV"
  >
    Type
    :
    crypto
  </div>
  <div
    className="Item__ItemWrapper-w49u1w-1 joGTJV"
  >
    Value
    :
    1
  </div>
  <div
    className="Item__ItemWrapper-w49u1w-1 joGTJV"
  >
    Unit
    :
    BTC
  </div>
</div>
`;
